aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -12,17 +12,13 @@ NestedTensorImpl::NestedTensorImpl(
     at::Tensor buffer,
     at::Tensor nested_size_tensor)
     : TensorImpl(
-          // TODO: This doesn't properly report is_cpu/is_cuda for NestedTensor.
-          // The intended resolution is that once #72827 lands we will be able to
-          // allocate separate dispatch keys for CPUNestedTensor (and any other
-          // hypothetical device backends for NestedTensor); then we will be
-          // able to derive this directly.  If you need this to work before then,
-          // make sure you add CPU to this dispatch key set
-          c10::DispatchKeySet({DispatchKey::NestedTensor}),
+          (c10::DispatchKeySet(DispatchKey::NestedTensor) |
+           c10::DispatchKeySet(buffer.is_cuda() ? BackendComponent::CUDABit : BackendComponent::CPUBit)),
```

Comment:
alternative would be to make the switch statement on `DispatchKey::NestedTensorCPU` vs. ` DispatchKey::NestedTensorCUDA`, but this works too :) Honestly I'm not sure which one is clearer to people (using "runtime keys" vs "building block keys" to make the keyset)